builder/solver_monitor.go

```diff
@@ -317,6 +317,9 @@ Loop:
 		}
 	}
 	if errVertex != nil {
+		v := ctx.Value("failed_output")
+		ptr, _ := v.(*string)
+		*ptr = string(errVertex.tailOutput.Bytes())
```

Comment:
This is too hacky.... :-)

Here's an alternative: Save it as a field in solver monitor and have a getter fun for it in builder. On failure, `buildAction` in `main.go` saves the output in a field as part of `app` (eg `lastFailureOutput`), which can be accessed from `run`.

Reply:
haha, why thank you :)

I was considering adding it to the solvermonitor; do you think it makes sense to then propigate it all the way back to where `app.cliApp.RunContext` is called? We're going to need to reference the builder object which gets created under  `actionBuild`. would it make sense to store an instance of the builder object under the `earthlyApp` struct?

Another somewhat (maybe less) hacky idea would be to have the builder return a `BuilderError` type, which we could create which would include both the underlying error as well as another field with the errVertex output.

I'm definitely up for suggestions.

Reply:
I was thinking that after an err is returned from builder in `actionBuild`, we can retrieve the output through a getter in builder that has access to `sm`. Then I was thinking of putting that output in a var in `app` that we can access from `run`.

A `BuilderError` that embeds the output too could be very elegant. One thing to be careful though is that sometimes errors are passed to buildkit and then buildkit gives it back to us - but it's in string form - the type of error is lost (it's wrapped in rpc stuff - we only get the string contents). So the error needs to be created in the right place in order for it to get to the caller without going through rpc.

Reply:
I had to do a combination of the two due to buildkit bubbling up the original error as a string; I created a `BuildError` (so I don't get linting issues with Builder.Builder* stuttering.